### PR TITLE
Progressive Web App support

### PR DESCRIPTION
**Pull request by HarryPeach**
Progressive Web Apps are web apps that have special javascript service workers.

 In this PR the site will be: 

- "Installable" on mobile devices from Chrome.
- Usable entirely offline
- Able to send push requests if that's required at any point

More info on PWAs can be found [on this Google-run site](https://web.dev/what-are-pwas/) if you aren't familiar (you might notice that this site is installable too).

So far, only basic support is added, and I still have to do a few things like:

- Change theme colors
- Attach icons for use on desktop / mobile
- A few other things that I'm forgetting

I'm open to any comments or suggestions on this :)

### CODE REVIEW

**Comment by theRTC204** [comment]
Please adjust whatever you used to lint the files to use spaces instead of tabs. This PR introduces an unnecessary entire white space diff.

**Comment by HarryPeach** [comment]
Oops, I'll get that sorted in the next commit

**Comment by HarryPeach** [comment]
This is mostly functional now, you can see a sample of the PWA deployed at https://wonderful-newton-76a770.netlify.com/

File: index.html [hunk 1]
```diff
@@ -1,293 +1,490 @@
...
-          <use xlink:href="#gentle-wave" x="48" y="7" fill="#fff" />
-        </g>
-      </svg>
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>Animal Crossing - Turnip Prophet</title>
+    <meta name="description" content="Animal Crossing - Turnip Prophet" />
+    <meta name="author" content="Mike Bryant" />
+    <meta name="theme-color" content="#00ff00" />
```
Comment (noahm) [inline index.html:10]: Shouldn't this match the theme color declared in the manifest? (or should it just be removed?)

**Comment by noahm** [comment]
Really looking forward to this merging. I clicked over from the app specifically looking for a PR adding this!

File: service-worker.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+// PWA Code adapted from https://github.com/pwa-builder/PWABuilder
+const CACHE = "pwa-precache";
+const precacheFiles = [
```
Comment (noahm) [inline service-worker.js:3]: For the app to be fully usable offline we will need a locally hosted & cached version of jquery as well.

File: index.html (hunk 1 above)
Comment (HarryPeach) [reply index.html:10]: That's one of the things that I was forgetting apparently :^)

**Comment by mikebryant** [comment]
> All rebased

This isn't rebased - there's loads of extra commits in there that are duplicates of others, and merge commits and so on. A `git rebase` on the upstream master should remove all of those.

**Comment by mikebryant** [comment]
@HarryPeach Thank you!

**Pull request merged**